tenants-attributes-checker/src/main/scala/it/pagopa/interop/tenantsattributeschecker/util/jobs.scala

```diff
@@ -0,0 +1,170 @@
+package it.pagopa.interop.tenantsattributeschecker.util
+
+import it.pagopa.interop.certifiedMailSender.InteropEnvelope
+import it.pagopa.interop.commons.utils.TypeConversions.{OptionOps, _}
+import it.pagopa.interop.commons.utils.service.UUIDSupplier
+import it.pagopa.interop.tenantmanagement.model.tenant.{
+  PersistentTenant,
+  PersistentTenantVerifier,
+  PersistentVerificationRenewal,
+  PersistentVerifiedAttribute
+}
+import it.pagopa.interop.tenantsattributeschecker.ApplicationConfiguration.{
+  actorSystem,
+  blockingEc,
+  certifiedMailQueueName,
+  context,
+  selfcareV2ApiKey,
+  selfcareV2URL
+}
+import it.pagopa.interop.tenantsattributeschecker.service._
+import it.pagopa.interop.tenantsattributeschecker.service.impl._
+import it.pagopa.interop.tenantsattributeschecker.util.errors._
+
+import java.util.UUID
+import scala.concurrent.Future
+
+object jobs {
+
+  private val agreementProcess: AgreementProcessService                 = AgreementProcessServiceImpl(blockingEc)
+  private val tenantProcess: TenantProcessService                       = TenantProcessServiceImpl(blockingEc)
+  private val attributeRegistryProcess: AttributeRegistryProcessService =
+    AttributeRegistryProcessServiceImpl(blockingEc)
+  private val queueService: QueueService                                = new QueueServiceImpl(certifiedMailQueueName)
+  private val selfcareClientService: SelfcareClientService              =
+    new SelfcareClientServiceImpl(selfcareV2URL, selfcareV2ApiKey)
+  private val (consumerExpiredTemplate, producerExpiredTemplate): (MailTemplate, MailTemplate)   = MailTemplate.expired
+  private val (consumerExpiringTemplate, producerExpiringTemplate): (MailTemplate, MailTemplate) =
+    MailTemplate.expiring
+
+  def applyStrategyOnExpiredAttributes(tenants: List[PersistentTenant]): Future[Unit] = {
+
+    Future
+      .traverse(tenants) { tenant =>
+        Future.traverse(tenant.attributes.collect { case v: PersistentVerifiedAttribute => v }) { attribute =>
+          Future.traverse(attribute.verifiedBy) { verifiedBy =>
```

Comment:
In this function and the following one a single traverse might be enough. I'll show you how in a refactoring session.